extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -292,6 +292,133 @@ export class ProjectsController {
 		this.refreshProjectsTree();
 	}
 
+	/**
+	 * Adds a database reference to the project
+	 * @param treeNode a treeItem in a project's hierarchy, to be used to obtain a Project
+	 */
+	public async addDatabaseReference(treeNode: BaseProjectTreeItem): Promise<void>;
+	/**
+	 * Adds a database reference to the project
+	 * @param project Project to be built and deployed
+	 */
+	public async addDatabaseReference(project: Project): Promise<void>;
+	public async addDatabaseReference(context: Project | BaseProjectTreeItem): Promise<void> {
```

Comment:
it seems to me you only need the third method signature, it covers the other 2 signatures.